pkg/api/scylla/validation/cluster_validation.go

```diff
@@ -221,6 +224,35 @@ func ValidateRepairTaskSpec(repairTaskSpec *scyllav1.RepairTaskSpec, fldPath *fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("intensity"), repairTaskSpec.Intensity, "must be a float"))
 	}
 
+	allErrs = append(allErrs, ValidateSchedulerTaskSpec(&repairTaskSpec.SchedulerTaskSpec, fldPath)...)
+
+	return allErrs
+}
+
+func ValidateBackupTaskSpec(backupTaskSpec *scyllav1.BackupTaskSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, ValidateSchedulerTaskSpec(&backupTaskSpec.SchedulerTaskSpec, fldPath)...)
+
+	return allErrs
+}
+
+func ValidateSchedulerTaskSpec(schedulerTaskSpec *scyllav1.SchedulerTaskSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// We can only validate interval when cron is non-nil for backwards compatibility.
+	if schedulerTaskSpec.Interval != nil && schedulerTaskSpec.Cron != nil {
+		if intervalDuration, err := duration.ParseDuration(*schedulerTaskSpec.Interval); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("interval"), schedulerTaskSpec.Interval, "valid units are d, h, m, s"))
+		} else if intervalDuration != 0 {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("interval"), "can't be non-zero when cron is specified"))
+		}
+	}
+
+	if schedulerTaskSpec.Timezone != nil && schedulerTaskSpec.Cron == nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("timezone"), "can't be set when cron is not specified"))
+	}
+
```

Comment:
please add timezone and cron format validation

Reply:
@tnozicka Scylla Manager accepts the `timezone` parameter, which, at least according to sctool docs, represents `Timezone of –cron and –window flag values`.

At the same time, it accepts `TZ` and `CRON_TZ` prefixes of `cron` parameter (see https://github.com/robfig/cron/blob/e843a09e5b2db454d77aad25b1660173445fb2fc/parser.go#L95).

The combination of two is accepted. K8s has a similar issue with CronJobs, and it doesn't accept TZ/CRON_TZ prefixes: see https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#unsupported-timezone-specification.

Should we handle it the same way k8s does? Should I raise this with the manager team?

Reply:
I think so + stricter is always better start with the validation

Reply:
ok, done

Reply:
Xref https://github.com/scylladb/scylla-manager/issues/3818